core/common/src/main/java/alluxio/util/JvmPauseMonitor.java

```diff
@@ -0,0 +1,195 @@
+  private static final Log LOG = LogFactory.getLog(
+      JvmPauseMonitor.class);
+
+  /** The time to sleep. */
+  private final long mGcSleepIntervalMs;
+
+  /** Extra sleep time longer than this threshold, log WARN. */
+  private final long mWarnThresholdMs;
+
+  /** Extra sleep time longer than this threshold, log INFO. */
+  private final long mInfoThresholdMs;
+
+  /** Times extra sleep time exceed WARN. */
+  private long mExceedWarnTimes = 0;
+  /** Times extra sleep time exceed INFO. */
+  private long mExceedInfoTimes = 0;
+  /** Total extra sleep time. */
+  private long mTotalExtraTime = 0;
+
+  private Thread mJvmMonitorThread;
+  private volatile boolean mThreadStarted = true;
+
+  /**
+   * Constructs JvmPauseMonitor.
+   */
+  public JvmPauseMonitor() {
+    mGcSleepIntervalMs = Configuration.getLong(PropertyKey.JVM_MONITOR_SLEEP_INTERVAL_MS);
+    mWarnThresholdMs = Configuration.getLong(PropertyKey.JVM_MONITOR_WARN_THRESHOLD_MS);
+    mInfoThresholdMs = Configuration.getLong(PropertyKey.JVM_MONITOR_INFO_THRESHOLD_MS);
+  }
+
+  /**
+   * Starts jvm monitor thread.
+   */
+  public void start() {
+    Preconditions.checkState(mJvmMonitorThread == null,
+        "JVM monitor thread already started");
+    mJvmMonitorThread = new Daemon(new Monitor());
+    mJvmMonitorThread.start();
+  }
+
+  /**
+   * Stops jvm monitor.
+   */
+  public void stop() {
+    mThreadStarted = false;
+    mJvmMonitorThread.interrupt();
+    try {
+      mJvmMonitorThread.join();
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+    }
+    reset();
+  }
+
+  /**
+   * Resets value of mJvmMonitorThread and mThreadStarted.
+   * @return reseted mThreadStarted
+   */
+  public JvmPauseMonitor reset() {
+    mJvmMonitorThread = null;
+    mThreadStarted = true;
+    return this;
+  }
+
+  /**
+   * @return true if thread started,false otherwise
+   */
+  public boolean isStarted() {
+    return mJvmMonitorThread != null;
+  }
+
+  /**
+   * @return Times exceed WARN threshold
+   */
+  public long getExceedWarnTimes() {
+    return mExceedWarnTimes;
+  }
+
+  /**
+   * @return Times exceed INFO threshold
+   */
+  public long getExceedInfoTimes() {
+    return mExceedInfoTimes;
+  }
+
+  /**
+   * @return Total extra time
+   */
+  public long getTotalExtraTime() {
+    return mTotalExtraTime;
+  }
+
+  private String getMemoryInfo() {
+    Runtime runtime = Runtime.getRuntime();
+    return "max memory = " + runtime.maxMemory() / (1024 * 1024) + "M total memory = "
+        + runtime.totalMemory() / (1024 * 1024) + "M free memory = "
+        + runtime.freeMemory() / (1024 * 1024) + "M";
+  }
+
+  private String formatLogString(long extraSleepTime,
+      List<GarbageCollectorMXBean> gcMXBeanListBeforeSleep,
+        List<GarbageCollectorMXBean> gcMXBeanListAfterSleep) {
+    List<String> diffBean = Lists.newArrayList();
+    GarbageCollectorMXBean oldBean;
+    GarbageCollectorMXBean newBean;
```

Comment:
Will the `oldBean` and `newBean` have the same elements in the same order?

Reply:
@yupeng9  yes,I made a test, and elements' order is the same.

Reply:
Add a defensive check on the bean name of before and after in the loop, in case any bean is added/removed during sleep.